vllm/entrypoints/llm.py

```diff
@@ -281,8 +282,7 @@ def generate(
             considered legacy and may be deprecated in the future. You should
             instead pass them via the ``inputs`` parameter.
         """
-        if self.llm_engine.model_config.embedding_mode or \
-            self.llm_engine.model_config.simple_mode:
+        if self.llm_engine.model_config.embedding_mode:
```

Comment:
We should still check `simple_mode` here imo. Perhaps we should make the mode an enum to avoid complex if-else cases as we add more modes..

Reply:
Good idea, Let me try.

Reply:
I created a class ModelMode to indicate the model mode [ENCODER, DECODER, EMBEDDING, SIMPLE]. Please take a look and give me some suggestions :)

Reply:
Added a new round of comments!